test/boilerplate_suggestors/props_and_state_classes_migrator_test.dart

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:over_react_codemod/src/boilerplate_suggestors/boilerplate_utilities.dart';
+import 'package:over_react_codemod/src/boilerplate_suggestors/props_and_state_classes_migrator.dart';
+import 'package:test/test.dart';
+
+import '../util.dart';
+
+main() {
+  group('PropsAndStateClassesMigrator', () {
+    SuggestorTester testSuggestor;
+
+    setUpAll(() async {
+      final helper = SemverHelper();
+      await helper.fromReport(
+          '/Users/sydneyjodon/Documents/GitHub/over_react_codemod/lib/src/boilerplate_suggestors/test.json');
```

Comment:
We need a version of the report tracked in Git somehow, as opposed to this hard-coded API.

To make testing easier, perhaps having a constructor that takes in the map directly would be helpful. That way, you can embed the JSON in the test, and also choose how to load the report in actual usage.

```dart
SemverHelper.fromReport(this.exportList);
```

```dart
// in the test for SemverHelper
final helper = SemverHelper({'exports': ...});
```

```dart
// in usage:
final helper = SemverHelper(jsonDecode(await File('report.json').readAsString()));
```

Reply:
Also, a test for a Suggestor would be a case where we'd want to use the mock for SemverHelper as opposed to a real instance, to make test setup easier (by not having to set up the whole exports map).